### PR TITLE
ENH: Add utility function to generate BIDS directory for testing

### PR DESCRIPTION
**Pull request by mgxd**
Ideally will port to pybids once somewhat mature / stable.

### CODE REVIEW

File: nibabies/utils/bids.py [hunk 1]
```diff
@@ -304,3 +304,79 @@ def validate_input_dir(exec_env, bids_dir, participant_label):
...
+
+
+def generate_bids_skeleton(location, _bids_dict):
+    """
+    Converts a BIDS directory in dictionary form to a file structure.
+    """
+    from copy import deepcopy
+
+    # ensure dictionary remains unaltered
+    bids_dict = _bids_dict.copy()
```
Comment (effigies) [inline nibabies/utils/bids.py:316]: Private variables as parameters don't really make sense. Also, I think you want `deepcopy`:

```Python
In [1]: from copy import deepcopy

In [2]: a = {"a": ["b"], "c": ["d"]}

In [3]: b = a.copy()

In [4]: c = deepcopy(a)

In [5]: b["a"].append("e")

In [6]: c["a"].append("f")

In [7]: a
Out[7]: {'a': ['b', 'e'], 'c': ['d']}

In [8]: b
Out[8]: {'a': ['b', 'e'], 'c': ['d']}

In [9]: c
Out[9]: {'a': ['b', 'f'], 'c': ['d']}
```

```suggestion
def generate_bids_skeleton(location, bids_dict):
    """
    Converts a BIDS directory in dictionary form to a file structure.
    """
    from copy import deepcopy

    # ensure dictionary remains unaltered
    bids_dict = deepcopy(bids_dict)
```

**Review by effigies — APPROVED** [review]
Small quibble.

**Comment by mgxd** [comment]
@effigies i think this would be better suited in either niworkflows / pybids, wdyt?

**Comment by effigies** [comment]
I agree. I guess it doesn't look like it's supporting any tests at this point... I would say let's keep this PR open so that we can merge it in and use it if needed. If we get it in an upstream release before we actually write a test using it, we can just close it.

**Review by mgxd — COMMENTED** [review]
I think this would be useful for testing https://github.com/nipreps/sdcflows/pull/257 - i'll likely port this into niworkflows

**Pull request closed without merging**